SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp

```diff
@@ -70,3 +74,80 @@ int32 USpatialNetConnection::IsNetReady(bool Saturate)
 	// This will always return true until we solve this.
 	return true;
 }
+
+void USpatialNetConnection::InitHeartbeat(FTimerManager* InTimerManager, Worker_EntityId InPlayerControllerEntity)
+{
+	checkf(PlayerControllerEntity == SpatialConstants::INVALID_ENTITY_ID, TEXT("InitHeartbeat: PlayerControllerEntity already set: %lld. New entity: %lld"), PlayerControllerEntity, InPlayerControllerEntity);
+	PlayerControllerEntity = InPlayerControllerEntity;
+	TimerManager = InTimerManager;
+
+	bIsServer = Driver->IsServer();
+
+	if (Driver->IsServer())
+	{
+		SetHeartbeatTimeoutTimer();
+
+		// Set up heartbeat event callback
+		TWeakObjectPtr<USpatialNetConnection> ConnectionPtr = this;
+		Cast<USpatialNetDriver>(Driver)->Receiver->AddHeartbeatDelegate(PlayerControllerEntity, HeartbeatDelegate::CreateLambda([ConnectionPtr](Worker_ComponentUpdateOp& Op)
+		{
+			if (ConnectionPtr.IsValid())
```

Comment:
I believe it could happen as a race condition of the client lagging for a while, then sending its heartbeat, but the server has just cleaned up the connection after deciding that the client has disconnected.